### PR TITLE
feat(esl-event-listener): add event delegated target

### PR DESCRIPTION
**Pull request by abarmina**
Closes: #1675

### CODE REVIEW

File: src/modules/esl-event-listener/README.md [hunk 1]
```diff
@@ -111,6 +111,27 @@ Here is the list of supported keys of `ESLEventDesriptor`:
   <u>Default Value:</u> `null`  
   <u>Description:</u> the CSS selector to filter event targets for event delegation mechanism.
 
+  ⚠ If you want to get the currently delegated event target, you can specify a new event type `DelegatedEvent<EventType>` for function and then the `$delegate` variable will contain the desired selector element.
+
```
Comment (dshovchko) [inline src/modules/esl-event-listener/README.md:115]:
```suggestion
  ⚠ If you want to get the currently delegated event target, you can specify a new event type `DelegatedEvent<EventType>` for the function and then the `$delegate` variable will contain the desired selector element.

```

File: src/modules/esl-event-listener/core/listener.ts [hunk 2]
```diff
@@ -103,18 +103,21 @@ export class ESLEventListener implements ESLListenerDefinition, EventListenerObj
...
       : handlerBound;
     return this.selector
-      ? (e: Event): void => this.isDelegatedTarget(e) && handlerFull(e)
+      ? (e: Event): void => this.handleDelegation(e, handlerFull)
       : handlerFull;
   }
 
-  /** Checks if the passed event can be handled by the current event listener */
-  protected isDelegatedTarget(e: Event): boolean {
+  /** Executes handler if the passed event accepted by the selector */
```
Comment (dshovchko) [inline src/modules/esl-event-listener/core/listener.ts:110]:
```suggestion
  /** Executes handler if the passed event is accepted by the selector */
```

File: src/modules/esl-event-listener/core/listener.ts [hunk 3]
```diff
@@ -103,18 +103,21 @@ export class ESLEventListener implements ESLListenerDefinition, EventListenerObj
...
+  protected handleDelegation(e: Event, handler: EventListener): void {
+    const delegate = this.delegate;
```
Comment (dshovchko) [inline src/modules/esl-event-listener/core/listener.ts:112]:
```suggestion
    const {delegate} = this;
```

File: src/modules/esl-event-listener/core/types.ts [hunk 4]
```diff
@@ -8,6 +8,11 @@ declare global {
   }
 }
 
+/** Event containing a delegated event target */
```
Comment (dshovchko) [inline src/modules/esl-event-listener/core/types.ts:11]:
```suggestion
/** Extended event with a delegated event target */
```

**Review by dshovchko — APPROVED** [review]

**Review by NastaLeo — APPROVED** [review]

File: src/modules/esl-event-listener/README.md (hunk 1 above)
Comment (abarmina) [reply src/modules/esl-event-listener/README.md]: at the suggestion of @ala-n, this part of the text was changed

**Review by ala-n — APPROVED** [review]

**Pull request merged**